api/v1/pathwaysjob_types.go

```diff
@@ -145,6 +153,30 @@ type ControllerSpec struct {
 	UserPodTemplate *corev1.PodTemplateSpec `json:"template,omitempty" protobuf:"bytes,6,opt,name=template"`
 }
 
+// CustomComponents struct lists the PathwaysJob attributes
+// that can be customized by the user.
+type CustomComponentsSpec struct {
+	// Image for the Pathways Resource Manager.
+	// +optional
+	PathwaysServerImage string `json:"pathwaysServerImage,omitempty"`
+
+	// Image for the Pathways Proxy.
+	// +optional
+	ProxyServerImage string `json:"proxyServerImage,omitempty"`
+
+	// Custom RM flags
+	// +optional
+	CustomPathwaysServerFlags []string `json:"customPathwaysServerFlags,omitempty"`
+
+	// Custom Proxy flags
+	// +optional
+	CustomProxyServerFlags []string `json:"customProxyServerFlags,omitempty"`
+
+	// Custom Worker flags
+	// +optional
+	CustomWorkerFlags []string `json:"customWorkerFlags,omitempty"`
```

Comment:
Future-proofing only, but can we also add custom flags for the RP container? We don't need them now but we could.

Reply:
Good thought, but I prefer to wait until we see this use-case. The API is extensible to add RP flags if needed.